### PR TITLE
Registrydisk rename

### LINKED ISSUE #1308: Rename registryDisk to containerDisk
Today we have a volume source called `registryDisk`.

1. The term itself implies that there is a disk registry, which is technically right, but might be misleading.
2. The API could lead to the impression that the whole container is used as the disk source for the VM. but in reality it is a file within the container which is used as a disk image.

These two points are worth to resolve, in order to not block us to actually eveolve other stories around registries and containers in the future (i.e. meaning of a real container in the vm spec, or other registries).

An alternative API could be along the lines of:

```yaml
kind: VM
spec:
  …
  volumes:
  - name: wooooah
    containerDisk:
      image: quay.io/bob/ubuntu:latest
      path: /disk.img
```

I could actuall yimagine that path defaults to a certain path, or that a glob is used by default. (i.e. first file in / or so).

### PR DESCRIPTION
**Pull request by petrkotas**
**What this PR does / why we need it**:
Rework the RegistryDisk to ContainerDisk according to issue #1308.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1308 

**Special notes for your reviewer**:

**Release note**:

```release-note
Rename RegistryDisk to ContainerDisk
```

---
This change is [Reviewable](https://reviewable.io/reviews/kubevirt/kubevirt/1730)

### CODE REVIEW

**Comment by petrkotas** [comment]
Awww, I forgot to push last commit, will do it first thing in the morning. Thanks for pointing this out @slintes

**Comment by fabiand** [comment]
@petrkotas please note in the docs that the path is optional, and what it defaults to by default.

Also: Is there some code in place to automatically convert from registryDisk to containerDisk if registryDisk is still used in a VMI spec?
If not,t hen we should get such a migration helper in palce.

**Comment by petrkotas** [comment]
@fabiand updated the user-guide, thanks for reviewing it :).

Regarding the migration, we either say this is a breaking change (which I assumed we agreed upon) or not. If not, than yes maintaining both APIs is an option. However, this will lead to confusion and will allow users to choose between `containerDisk` and `registryDisk`, or even worse, use them together in one VMI.

I personally does not vouch for the path of maintaining both. Lets to the breaking change and announce it with other breaking changes we did. This change is made to clear the confusion, and not create additional with both option available.

File: pkg/api/v1/schema.go [hunk 1]
```diff
@@ -439,6 +439,8 @@ type ContainerDiskSource struct {
 	Image string `json:"image"`
 	// ImagePullSecret is the name of the Docker registry secret required to pull the image. The secret must already exist.
 	ImagePullSecret string `json:"imagePullSecret,omitempty"`
+	// Path defines the path to disk file in the container
+	Path string `json:"path"`
```
Comment (slintes) [inline pkg/api/v1/schema.go]: since this is optional, can you add `omitempty`?

Comment (petrkotas) [reply pkg/api/v1/schema.go]: Good point, done.

**Comment by fabiand** [comment]
There is no question that we want this change, and we gave head's up that
there will be breaking changes, thus I am okay to get this merged now.

If we want to provide - for convenience - a migration path, then we can
treat this as a separate feature and provide it as a follow up.

**Comment by slintes** [comment]
thanks for clarification Fabian, will merge

**Pull request merged**